igvc/src/imu/main.cpp

```diff
@@ -91,29 +99,56 @@ int main(int argc, char** argv)
 
     try
     {
-      msg.linear_acceleration.x = stof(accelTokens[0]);
-      msg.linear_acceleration.y = stof(accelTokens[1]);
-      msg.linear_acceleration.z = stof(accelTokens[2]);
+      double cur_x = stof(accelTokens[0]);
+      double cur_y = stof(accelTokens[1]);
+      double cur_z = stof(accelTokens[2]);
       msg.linear_acceleration_covariance = { 0.04, 1e-6, 1e-6, 1e-6, 0.04, 1e-6, 1e-6, 1e-6, 0.04 };
 
+      int num = 500;
+
+      x_accel.push_back(cur_x);
+      if(x_accel.size() > num) {
+        double front = x_accel.front();
+        x_accel.pop_front();
+        x_avg -= front / num;
+      }
+      x_avg += cur_x / num;
+
+      z_accel.push_back(cur_z);
+      if(z_accel.size() > num) {
+        double front = z_accel.front();
+        z_accel.pop_front();
+        z_avg -= front / num;
+      }
+      z_avg += cur_z / num;
+
+      y_accel.push_back(cur_y);
+      if(y_accel.size() > num) {
+        double front = y_accel.front();
+        y_accel.pop_front();
+        y_avg -= front / num;
+      }
+      y_avg += cur_y / num;
+
+      msg.linear_acceleration.x = cur_x - x_avg;
+      msg.linear_acceleration.y = cur_y - y_avg;
+      msg.linear_acceleration.z = cur_z - z_avg;
+
+
       msg.angular_velocity.x = stof(gyroTokens[0]);
       msg.angular_velocity.y = stof(gyroTokens[1]);
       msg.angular_velocity.z = stof(gyroTokens[2]);
       msg.angular_velocity_covariance = { 0.02, 1e-6, 1e-6, 1e-6, 0.02, 1e-6, 1e-6, 1e-6, 0.02 };
 
-      float yaw = HALF_TO_FULL_CIRCLE_ANGLE(-(stof(rpyTokens[0]) + 90)) * DEG_TO_RAD;
+      float yaw = HALF_TO_FULL_CIRCLE_ANGLE(-(stof(rpyTokens[0]) - 100)) * DEG_TO_RAD;
```

Comment:
Why was `+ 90` changed to `- 100`?  If there's some sort of tuning going on here, I think it should be written similar to `- 90 - tuning_constant`